packages/core/src/test/shared/env/resolveEnv.test.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import * as resolveEnv from '../../../shared/env/resolveEnv'
+import sinon from 'sinon'
+import path from 'path'
+
+describe('resolveEnv', async function () {
+    let sandbox: sinon.SinonSandbox
+    beforeEach(function () {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    // a copy of resolveEnv.mergeResolvedShellPath for stubbing
+    const testMergeResolveShellPath = async function mergeResolvedShellPath(
```

Comment:
why not stub `resolveEnv.getResolvedShellEnv` directly? that would be worth explaining (concisely) in the above comment

Reply:
`mergeResolvedShellPath` is calling `getResolvedShellEnv` within the same file (with out importing it). Is there a way for us to make `mergeResolvedShellPath` call the stub instead? I didn't figure that out before

Reply:
Added this in comment